File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant: list, completion: list):
    ## 한명을 찾아야하는데 일단 그렇담?
    ## 딕셔너리를 써볼까 ? 그냥 빼도 나올 듯하기도하고 ..?
    dic = {}
    for p in participant:
        if p not in dic:
            dic[p]=1
        else:
            dic[p]=dic.get(p)+1
    for c in completion:
        dic[c]=dic.get(c)-1
        
    answer =''
    for key in dic.keys():
        if dic[key] != 0:
            answer=key
            break

    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {

    unordered_map<string, int> um;
    for (string p : participant) {
        if (um.find(p) == um.end()) {
            um[p] = 1;
        }else {
            um[p]++;
        }
    }
    
    for (string s : completion) {
        um[s]--;
    }

    for (string p : participant) {
        if (um[p] != 0) {
            return p;
        }
    }
    
}